examples/role-labels/gen-go/role/ringpop-role.go

```diff
@@ -0,0 +1,159 @@
+package role
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/uber/ringpop-go"
+	"github.com/uber/ringpop-go/forward"
+	"github.com/uber/ringpop-go/router"
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/thrift"
+)
+
+type RingpopRoleServiceAdapter struct {
+	impl    TChanRoleService
+	ringpop ringpop.Interface
+	ch      *tchannel.Channel
+	config  RoleServiceConfiguration
+	router  router.Router
+}
+
+// RoleServiceConfiguration contains the forwarding configuration for the RoleService service. It has a field for every endpoint defined in the service. In this field the endpoint specific forward configuration can be stored. Populating these fields is optional, default behaviour is to call the service implementation locally to the process where the call came in.
+type RoleServiceConfiguration struct {
+	// GetMembers holds the forwarding configuration for the GetMembers endpoint defined in the service
+	GetMembers *RoleServiceGetMembersConfiguration
+	// SetRole holds the forwarding configuration for the SetRole endpoint defined in the service
+	SetRole *RoleServiceSetRoleConfiguration
+}
+
+func (c *RoleServiceConfiguration) validate() error {
+	if c.GetMembers != nil {
+		if c.GetMembers.Key == nil {
+			return errors.New("configuration for endpoint GetMembers is missing a Key function")
+		}
+	}
+	if c.SetRole != nil {
+		if c.SetRole.Key == nil {
+			return errors.New("configuration for endpoint SetRole is missing a Key function")
+		}
+	}
+	return nil
+}
+
+// NewRingpopRoleServiceAdapter creates an implementation of the TChanRoleService interface. This specific implementation will use to configuration provided during construction to deterministically route calls to nodes from a ringpop cluster. The channel should be the channel on which the service exposes its endpoints. Forwarded calls, calls to unconfigured endpoints and calls that already were executed on the right machine will be passed on the the implementation passed in during construction.
```

Comment:
This is part of generated code, I'd rather not change that at this point. Will make a branch to see if it helps to add a comment to the top of the generated code indicating it is generated so github doesn't show the contents